### ISSUE TITLE
Network Error: Connection reset by peer (os error 104)

### ISSUE THREAD

**Issue by michel-thomas** [orf/git-workspace#363]
I don't use often `git-workspace` but since one or two month I get this error:
```bash
$ git workspace update
Fetching repositories...
Error: Error fetching repositories from Gitlab user/group michel-thomas at https://gitlab.com in directory gitlab.com, using the token stored in GITLAB_TOKEN

Caused by:
    0: https://gitlab.com/api/graphql: Network Error: Network Error: Error encountered in the status line: Connection reset by peer (os error 104)
    1: Network Error: Error encountered in the status line: Connection reset by peer (os error 104)
    2: Connection reset by peer (os error 104)
```

I checked the token: Ok, scope `api`, not expired.

When I go into one of my repos subdir, `git pull` / `git fetch` is working well.

And version is latest:
```bash
$ git workspace -V
git-workspace 1.6.0
```

Thanks for your help.

**Comment by jagu-sayan** [comment]
Hello @michel-thomas,

Do you use some custom ssl certificates or are you behind a firewall or proxy ?
Maybe it's related to https://github.com/orf/git-workspace/pull/343

**Comment by michel-thomas** [comment]
No, I run it on my laptop Ubuntu 24.04 behind my ISP. And GitLab.com has its own TLS certificates.

I use SSH with ed25519 key, and testing connection is ok:
```bash
$ ssh -T git@gitlab.com        
Welcome to GitLab, @michel-thomas!
```

**Comment by jagu-sayan** [comment]
Maybe your GITLAB_TOKEN is expired ? I think that when I see this error `Error fetching repositories from Gitlab user/group michel-thomas at https://gitlab.com in directory gitlab.com, using the token stored in GITLAB_TOKEN`

`git-workspace` try to do a graphql request using your GITLAB_TOKEN to fetch repositories in your Gitlab group.

The rest of the error looks more like a network error, but maybe this error is not handled gracefully, I will check that.
